### PR TITLE
Oleksijt task4 define a base class for auto

### PR DESCRIPTION
**Pull request by Oleksijt**
Please check my code with classes for autos.

### CODE REVIEW

File: oleksijt/task4_define_a_base_class_for_auto.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+class Car:
+    """ Base class for autos. """
+    def __init__(self, car_type, model, year, max_speed):
+        self.car_type = car_type
+        self.model = model
+        self.year = year
+        self.max_speed = max_speed
+
+    def change_type(self, car_type):
+        self.__class__ = eval(car_type.title())  # Changes class.
```
Comment (pivancy) [inline oleksijt/task4_define_a_base_class_for_auto.py]: I don't recommend to use `eval` function.
In more cases, it's not secure.

Comment (Oleksijt) [reply oleksijt/task4_define_a_base_class_for_auto.py]: Eval removed. Changed to gettatr.

File: oleksijt/task4_define_a_base_class_for_auto.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    def __init__(self, car_type, model, year, max_speed):
+        self.car_type = car_type
+        self.model = model
+        self.year = year
+        self.max_speed = max_speed
+
+    def change_type(self, car_type):
+        """Change class and car_type. If class doesn't exist,
+        class sets as base class."""
+        self.__class__ = getattr(sys.modules[__name__], car_type.title(), Car)
```
Comment (pivancy) [inline oleksijt/task4_define_a_base_class_for_auto.py]: It's not very good idea to change `__class__` attribute.

Comment (Oleksijt) [reply oleksijt/task4_define_a_base_class_for_auto.py]: But othervise if we change car_type without  class, object will have methods and parameters of old class. For example class Electro may have parameter distance_per_charge.
Should i remove class changing?

Comment (pivancy) [reply oleksijt/task4_define_a_base_class_for_auto.py]: Yeap.

Comment (pivancy) [reply oleksijt/task4_define_a_base_class_for_auto.py]: For this reason I'll try to add a separate task.

Comment (Oleksijt) [reply oleksijt/task4_define_a_base_class_for_auto.py]: Done.

**Comment by pivancy** [comment]
👍

**Pull request merged**